### ISSUE TITLE
[leo_object_storage][test] Cover issues found during 1.3.1 development

### ISSUE THREAD

**Issue by mocchira** [leo-project/leofs#530]
As lots of issues about compaction found during 1.3.1 development cycle,
I'd like to add tests to cover those.

- [x] #520 (add huge garbage blocks and check properly skipping within reasonable time)
- [x] #521 (add check pid_pairs in leo_compact_fsm_controller is empty after every test finished)
- [x] #522 (add calls `sys:get_state/1` to leo_compact_fsm_workers running compaction|diagnose)
- [x] #523 (already covered by existing cases)
- [x] #524 (will be coverd by the check for #521) 
- [x] #526 (update the same record many times to cause leveldb start compaction and kick compact|diagnose afterwards) 
- [x] #527 (add invalid records which break one of the new rules and check properly skipping)
- [x] run in parallel with enough amount of tasks

**Comment by yosukehara** [comment]
@mocchira I've tested the latest `leo_object_storage` on our environment, which has been passed over ten straight times. Thanks for your contribution.

**Comment by mocchira** [comment]
@yosukehara great to hear that.
